app/services/plan_service/data_types.rb

```diff
@@ -9,6 +9,8 @@ module PlanService::DataTypes
     [:id, :fixnum, :optional], # For OS, the plan is not actually in DB. Thus, optional.
     [:community_id, :fixnum, :mandatory],
     [:plan_level, :fixnum, :mandatory],
+    [:features, :array, :optional],
```

Comment:
I'm thinking, is array the right data structure for features or should it be hash?

What if we have a feature which allows users to take backups from their marketplace in admin site. That would be a operation that takes a lot of time and thus we may want to limit the number of backups they can take per day. For starter plans, the value could be 1 backup / day and for pro plans it could be 6 backups / day. In that case, the array is not enough but hash would do the trick:

```
# trial
features: {
  whitelabel: false,
  backups_per_day: 0
}

# starter
features: {
  whitelabel: false,
  backups_per_day: 1
}

# pro
features: {
  whitelabel: true,
  backups_per_day: 6
}